src/main/java/jcifs/dcerpc/msrpc/lsarpc.java

```diff
@@ -0,0 +1,1046 @@
+                _src.advance(2 * _name_bufferl);
+
+                if ( this.name.buffer == null ) {
+                    if ( _name_buffers < 0 || _name_buffers > 0xFFFF )
+                        throw new NdrException(NdrException.INVALID_CONFORMANCE);
+                    this.name.buffer = new short[_name_buffers];
+                }
+                _src = _src.derive(_name_bufferi);
+                for ( int _i = 0; _i < _name_bufferl; _i++ ) {
+                    this.name.buffer[ _i ] = (short) _src.dec_ndr_short();
+                }
+            }
+            if ( _sidp != 0 ) {
+                if ( this.sid == null ) { /* YOYOYO */
+                    this.sid = new rpc.sid_t();
+                }
+                _src = _src.deferred;
+                this.sid.decode(_src);
+
+            }
+        }
+    }
+
+    public static class LsarDnsDomainInfo extends NdrObject {
+
+        public rpc.unicode_string name;
+        public rpc.unicode_string dns_domain;
+        public rpc.unicode_string dns_forest;
+        public rpc.uuid_t domain_guid;
+        public rpc.sid_t sid;
+
+
+        @Override
+        public void encode ( NdrBuffer _dst ) throws NdrException {
+            _dst.align(4);
+            _dst.enc_ndr_short(this.name.length);
+            _dst.enc_ndr_short(this.name.maximum_length);
+            _dst.enc_ndr_referent(this.name.buffer, 1);
+            _dst.enc_ndr_short(this.dns_domain.length);
+            _dst.enc_ndr_short(this.dns_domain.maximum_length);
+            _dst.enc_ndr_referent(this.dns_domain.buffer, 1);
+            _dst.enc_ndr_short(this.dns_forest.length);
+            _dst.enc_ndr_short(this.dns_forest.maximum_length);
+            _dst.enc_ndr_referent(this.dns_forest.buffer, 1);
+            _dst.enc_ndr_long(this.domain_guid.time_low);
+            _dst.enc_ndr_short(this.domain_guid.time_mid);
+            _dst.enc_ndr_short(this.domain_guid.time_hi_and_version);
+            _dst.enc_ndr_small(this.domain_guid.clock_seq_hi_and_reserved);
+            _dst.enc_ndr_small(this.domain_guid.clock_seq_low);
+            int _domain_guid_nodes = 6;
+            int _domain_guid_nodei = _dst.index;
+            _dst.advance(1 * _domain_guid_nodes);
+            _dst.enc_ndr_referent(this.sid, 1);
+
+            if ( this.name.buffer != null ) {
+                _dst = _dst.deferred;
+                int _name_bufferl = this.name.length / 2;
+                int _name_buffers = this.name.maximum_length / 2;
+                _dst.enc_ndr_long(_name_buffers);
+                _dst.enc_ndr_long(0);
+                _dst.enc_ndr_long(_name_bufferl);
+                int _name_bufferi = _dst.index;
+                _dst.advance(2 * _name_bufferl);
+
+                _dst = _dst.derive(_name_bufferi);
+                for ( int _i = 0; _i < _name_bufferl; _i++ ) {
+                    _dst.enc_ndr_short(this.name.buffer[ _i ]);
+                }
+            }
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 230:244, 245:259, 260:274, 739:753.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.